tools/retest/main.go

```diff
@@ -23,75 +26,180 @@ func main() {
 	handleError(err)
 	log.Printf("Found %d PRs", search.GetTotal())
 
+loop:
 	for _, pr := range search.Issues {
 		prNumber := pr.GetNumber()
-		log.Printf("Processing #%d", prNumber)
-		comments, _, err := client.Issues.ListComments(ctx, "stackrox", "stackrox", prNumber, nil)
+		log.Printf("#%d processing...", prNumber)
+		prDetails, _, err := client.PullRequests.Get(ctx, S, S, prNumber)
 		handleError(err)
+		commentsBodies := commentsForPR(ctx, client, prNumber)
+		log.Printf("#%d has %d comments", prNumber, len(commentsBodies))
+		checks := checksForCommit(ctx, client, prDetails.GetHead().GetSHA())
+		log.Printf("#%d has %d completed checks", prNumber, len(checks))
+
+		for name, status := range checks {
+			if !status {
+				log.Printf("#%d has a failing check (%s) skipping", prNumber, name)
+				continue loop
+			}
+		}
 
-		retested := 0
-		for _, c := range comments {
-			if c.GetBody() == retestComment {
-				retested++
+		statuses := statusForPR(ctx, client, prDetails.GetStatusesURL())
+		log.Printf("#%d has %d statuses", prNumber, len(statuses))
+		jobsToRetest := jobsToRetestFromComments(commentsBodies)
+		log.Printf("#%d jobs to retest: %s", prNumber, strings.Join(jobsToRetest, ", "))
+
+		for _, job := range jobsToRetest {
+			state := statuses[job]
+			if state == "pending" {
+				log.Printf("#%d %s is still pending", prNumber, job)
+				continue
 			}
+			log.Printf("#%d retesting: %s %s", prNumber, job, state)
+			comment := github.IssueComment{
+				Body: ptr("/test " + job),
+			}
+			testComment, _, err := client.Issues.CreateComment(ctx, S, S, prNumber, &comment)
+			handleError(err)
+			log.Printf("#%d commented: %s", prNumber, testComment.GetHTMLURL())
 		}
-		log.Printf("#%d was retested %d times", prNumber, retested)
-		if retested > 3 {
+
+		if len(jobsToRetest) != 0 {
 			continue
 		}
 
-		prDetails, _, err := client.PullRequests.Get(ctx, "stackrox", "stackrox", prNumber)
+		retestComment := github.IssueComment{
+			Body: ptr(retestComment),
+		}
+		if !shouldRetest(statuses, commentsBodies) {
+			continue
+		}
+		comment, _, err := client.Issues.CreateComment(ctx, S, S, prNumber, &retestComment)
 		handleError(err)
+		log.Printf("#%d commented: %s", prNumber, comment.GetHTMLURL())
+	}
+}
 
-		if isGHACheckFailing(ctx, client, prDetails.GetHead().GetSHA()) {
+var (
+	restestNTimes = regexp.MustCompile("Retest (.*) (\\d+) times")
```

Comment:
Let's use a consistent chatops language. How about:

```suggestion
	restestNTimes = regexp.MustCompile("/retest-times (\\d+) (.*)")
```